brainclient/brainClient.go

```diff
@@ -0,0 +1,87 @@
+package brainclient
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/bits-and-blooms/bloom/v3"
+)
+
+type BrainClient interface {
+	CheckUrl(url string) (bool, error)
+	DownloadBloomFilter() *bloom.BloomFilter
+}
+
+type Client struct {
+	brainAddress       string
+	indicatorsEndpoint string
+	detailsEndpoint    string
+}
+
+func NewClient(brainAddr string, indicatorsEndpoint string, detailsEndpoint string) *Client {
+	return &Client{
+		brainAddress:       brainAddr,
+		detailsEndpoint:    brainAddr + detailsEndpoint,
```

Comment:
maybe we can use an object like this to represent the address: https://pkg.go.dev/go.lsp.dev/uri

in this way the client is not costructed if something is wrong.. but at the same time i'm not 100% sure about this, since we will get an error when we will contact the cloud server for the first time